sumpy/symbolic.py

```diff
@@ -243,6 +244,33 @@ def find_power_of(base, prod):
     return result[power]
 
 
+class SpatialConstant(prim.Variable):
```

Comment:
I'd prefer if this had its own `mapper_method`, instead of piggybacking off of `map_variable` and using `isinstance`.

Reply:
What do you think about moving this to `pymbolic`? If we don't move this `pymbolic`, issue is that, I have to override a lot of mappers of pymbolic in both sumpy and pytential with `map_spatial_constant=map_variable`.